lib/widgets/course_detail_widgets.dart

```diff
@@ -105,39 +116,27 @@ class GroupDetails extends StatelessWidget {
 }
 
 class EventCalendar extends StatelessWidget {
-  final Schedule _schedule;
+  final ScheduleTranslate _schedule;
   const EventCalendar(this._schedule);
 
-  String _dateTimeToString(DateTime? dateTime) {
-    if (dateTime == null) return "unknown";
-
-    return dateTime.year.toString() +
-        " " +
-        dateTime.day.toString() +
-        "." +
-        dateTime.month.toString() +
-        ". " +
-        dateTime.hour.toString() +
-        ":" +
-        dateTime.minute.toString();
-  }
-
-  Row _eventToWidget(Event event) {
+  Row _eventToWidget(EventTranslate event) {
     return Row(
       children: [
-        Text(_dateTimeToString(event.begin) +
-            " - " +
-            _dateTimeToString(event.end)),
+        Text(event.getDate() +
+            " " +
+            event.getBeginHour() +
+            "-" +
+            event.getEndHour()),
         Spacer(),
-        Text(event.locations?[0].room ?? "unknown location")
+        Text(event.getRoom())
       ],
     );
   }
 
   Column _buildEventCalendar() {
     List<Row> view = [];
 
-    for (Event event in _schedule.events) {
+    for (EventTranslate event in _schedule.getEvents()) {
```

Comment:
Would love to get an overview of how this works, ping me when you have time so we can talk about it